include/psa/crypto_extra.h

```diff
@@ -1130,14 +1130,17 @@ typedef uint32_t psa_pake_primitive_t;
  * This is the proof in the Schnorr Non-Interactive Zero-Knowledge Proof (the
  * value denoted by the letter 'r' in RFC 8235).
  *
- * Both input and output at this step is an integer less than the group
- * selected in the cipher suite. The format depends on the group as well:
+ * Both for input and output, the value at this step is an integer less than
+ * the order of the group selected in the cipher suite. The format depends on
+ * the group as well:
  *
- * - For Montgomery curves, the encoding is little endian (see decoding of
- *   scalars in Section 5 of RFC 7748).
+ * - For Montgomery curves, the encoding is little endian.
  * - For everything else the encoding is big endian (see Section 2.3.8 of
  *   _SEC 1: Elliptic Curve Cryptography_ at https://www.secg.org/sec1-v2.pdf).
  *
+ * In both cases leading zeroes are allowed as long as the length in bytes does
+ * not exceed the byte length of the group order.
```

Comment:
That is true for secret values. These are public. They are very much like signatures: verifying a signature proves that the holder of the private key signed the message. Verifying the ZK proof proves that the holder has the key and nothing else.

Reply:
Ah, yes, good point, I was looking at this sentence in isolation and hadn't realized it only applied to public values.

Is there any advantage in allowing output to have leading zeros truncated though?

Reply:
> Is there any advantage in allowing output to have leading zeros truncated though?

Reduced cognitive load both on the user's and library implementor's part and maybe a marginal code size advantage since the implementation needs to do one less check.

Reply:
I think having a predictable data size is easier for both applications and implementations. But it's minor and we can change it later.

(I'm reminded of a security presentation: “The protocol included one variable-length field. So we tried a stupidly long input. The parser crashed.”)